src/unix/linux-core.c

```diff
@@ -1015,14 +1016,261 @@ uint64_t uv_get_total_memory(void) {
+#define CGROUPS_VERSION_2 0x2
+#define CGROUPS_V2_CTRL_PREFIX "0::"
+#define CGROUPS_V2_CTRL_PREFIX_LEN 3
+#define CGROUPS_V2_NO_LIMIT_VALUE "max"
+#define CGROUPS_BUF_SIZE 4096
+
+
+typedef struct {
+  /* cgroups version, one of CGROUPS_VERSION_*. */
+  uint8_t cgroups_version;
+  /* Path in which param files are found for a subsystem. */
+  char path[CGROUPS_BUF_SIZE];
+} uv__cgroups_subsystem_info_t;
+
+
+/*
+ * Writes `start` to `field_start` and the length of the current
+ * whitespace-delimited field to `field_len`, and returns an pointer to the
+ * remainder of the string (excluding the whitespace).
+ * Note that we assume that `start` ends with "\n\0".
+ */
+static const char* uv__mountinfo_next_field(const char* start,
+    const char** field_start, size_t* field_len) {
+  size_t len;
+  const char* field_end;
+  if ('\0' == *start)
+    return 0;
+  field_end = strchr(start, ' ');
+  if (field_end == NULL) {
+    len = strlen(start) - 1; /* -1 to lop off newline */
+  } else {
+    len = field_end - start;
+  }
+  if (field_start != NULL) {
+    *field_start = start;
+  }
+  if (field_len != NULL) {
+    *field_len = len;
+  }
+  return start + len + 1;
+}
+
+
+/*
+ * Given a comma-delimited sequence of strings `super_options` (w/ size
+ * `super_options_len`), return TRUE if option exactly matches one of the
+ * elements in that sequence.
+ */
+static int uv__mountinfo_find_super_option(const char* super_options,
+    size_t super_options_len, const char* option) {
+  size_t option_len = strlen(option);
+  const char* low = super_options;
+  const char* high = super_options + super_options_len - option_len;
+  const char* preceding_char;
+  const char* succeeding_char;
+  while (strlen(low)) {
+    low = strstr(low, option);
+    if (low == NULL || low > high)
+      break;
+    preceding_char = low - 1;
+    succeeding_char = low + option_len;
+    if (low == super_options || ',' == *preceding_char) {
+      /*
+       * The occurrence of `option` is preceded by a comma, or is part of the
+       * first entry in `super_options`.
+       */
+      if (' ' == *succeeding_char || ',' == *succeeding_char || '\n' == *succeeding_char)
+        /*
+         * The occurrence of `option` is succeeded by a comma, or is part of the
+         * last entry in `super_options`.
+         */
+        return 1;
+    }
+  }
+  return 0;
+}
+
+
+/*
+ * Read /proc/self/mountinfo and /proc/self/cgroup to get info about how the
+ * given subsystem is controlled for this process, and the path to the
+ * subsystem's parameter files, if possible.
+ */
+static int uv__read_cgroups_proc_files(uv__cgroups_subsystem_info_t* info, const char* subsystem) {
+  FILE* fp;
+  /* Should be big enough to fit a single line. */
+  char buf[CGROUPS_BUF_SIZE];
+  /*
+   * See description of /proc/[pid]/mountinfo in proc(5).
+   * Henceforth, numbers in parentheses refer to fields from the docs.
+   */
+  const char* field_ptr;
+  const char* field_start[10];
```

Comment:
Use named fields instead. Since you will be using strsep, they would be null terminated, so you don't need to have a `field_len` array.